python/fusion_engine_client/messages/gnss_corrections.py

```diff
@@ -0,0 +1,58 @@
+import math
+
+from construct import (Struct, Float32l, Int8ul, Int16ul, Int64sl, Padding, Array, this)
+
+from ..utils.construct_utils import construct_message_to_string
+from .defs import *
+
+
+class LBandFrameMessage(MessagePayload):
+    """!
+    @brief L-Band frame message.
+    """
+    MESSAGE_TYPE = MessageType.LBAND_FRAME
+    MESSAGE_VERSION = 0
+
+    LBandFrameMessageConstruct = Struct(
+        "system_time_ns" / Int64sl,
+        "user_data_size_bytes" / Int16ul,
+        "bit_error_count" / Int16ul,
+        "signal_power_db" / Int8ul,
+        Padding(3),
+        "doppler_hz" / Float32l,
+        "data_payload" / Array(this.user_data_size_bytes, Int8ul),
+    )
+
+    def __init__(self):
+        self.system_time_ns = 0
+        self.bit_error_count = 0
+        self.signal_power_db = 0
+        self.doppler_hz = math.nan
+        self.data_payload = bytes()
+
+    def pack(self, buffer: Optional[bytes] = None, offset: int = 0, return_buffer: bool = True) -> (bytes, int):
+        values = vars(self)
+        values['user_data_size_bytes'] = len(self.data_payload)
+        packed_data = self.LBandFrameMessageConstruct.build(values)
+        return PackedDataToBuffer(packed_data, buffer, offset, return_buffer)
+
+    def unpack(self, buffer: bytes, offset: int = 0, message_version: int = MessagePayload._UNSPECIFIED_VERSION) -> int:
+        parsed = self.LBandFrameMessageConstruct.parse(buffer[offset:])
+        self.__dict__.update(parsed)
+        del self.__dict__['_io']
```

Comment:
Also need to delete `user_data_size_bytes` to avoid adding a new class member. I'd suggest doing these deletes on the `parsed` dict before the update call so they never become members at all.

Reply:
You actually need _io to be there in parsed.